python/ray/data/_internal/batcher.py

```diff
@@ -78,3 +121,203 @@ def next_batch(self) -> Block:
         # blocks consumed on the next batch extraction.
         self._buffer = leftover
         return output.build()
+
+
+class ShufflingBatcher(BatcherInterface):
+    """Chunks blocks into shuffled batches, using a local in-memory shuffle buffer."""
+
+    # Implementation Note:
```

Comment:
Ah, let me change the `Batcher` implementation note to follow this pattern. I've always liked this documentation pattern of keeping docstrings focused on what's needed to know in order to use the abstraction/API (user-focused), and using block comments for implementation/design details that are only useful to abstraction maintainers rather than users (developer-focused).